fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/notification/NotificationCollector.java

```diff
@@ -0,0 +1,47 @@
+package org.technologybrewery.fermenter.mda.notification;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * An accessible class to make it easy to add notification from anywhere within the fermenter-mda classpath.
+ * <p>
+ * Theis collector support multi-threading and with notifications gathered and stored automatically across
+ * multi-module builds.
+ */
+public final class NotificationCollector {
+    private static final ThreadLocal<Map<String, Map<String, Notification>>> NOTIFICATIONS = ThreadLocal.withInitial(ConcurrentHashMap::new);
```

Comment:
Q: Is this any issue with multithreading support? I thought maybe a year or so ago we were pushing for thread safety to make the builds faster. Obviously this is threadsafe, but just wondering if we'd lose any messages on separate threads?

Reply:
Completely missed the concurrent hashmap! Thanks for confirming.